### ISSUE TITLE
Suggestion: Why wait for the burn-down?

### ISSUE THREAD

**Issue by bunnybot** [widelands/widelands#2801]
For some time I have been wondering why we're all forced to watch in awe how our buildings burn down,  when we want to dedicate the space to some other function.
I find this a bit annoying in a game that can get quite hectic at times, and I cannot see any deeper meaning in this.
It forces us to wait and watch, or, more often, to come back to the burning site after a while, and finally initiate what we wanted to do in the first place. After all we very often need to so that with lumberjacks, mines, fishermen's and quarries.
Would it not be beneficial for the gameplay, if we could burn the old building, see our options (build help icon space, roads)  and determine the further usage immediately, while the fires are still burning?
I do not suggest any principle change in the timing and sequence here. The constructions site would still be erected only after the fires have stopped. I only want the game to remember the player's commands and fulfil them once the place is cleared, so that destroying a house and building another one at the same place becomes a 1-step process.
In other words, playing the burn-down animation should not block the player's commands, but still delay their execution.

I do not include the case of buildings burning down due to having lost land here, because it may make matter more complicated. So the game should behave differently depending on whether the burning was initiated by a border change, or a direct player command. Only the latter should be subject of the change.
I know this change may be not trivial.

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2011-09-28T06:28:46Z
 * owner: wolfsteinmetz
 * the launchpad url was https://bugs.launchpad.net/bugs/861165

**Comment by bunnybot** [comment]
(by sirver)
Ahh, I feel we are here in the there-is-no-right-solution that borders on the question: what should the player do vs what should be automated. I feel that it is your responsibility to burn houses and return after the land became useful again to plant new buildings as both are controlling tasks. I understand that you see this different. Of course, I will now bring the argument that one could pull your argument to the absolute extreme: the computer know best how to place buildings in an optimized fashion - this is an optimization problem with a high dimensionality. So why should you have to place the buildings in the first place? I understand that this doesn't move the discussion forward. So basically, this bug report boils down to a vote for me... therefore

I am +1 for keeping the current state as is.

**Comment by bunnybot** [comment]
(by nasenbaer)
I completely agree with SirVer, (although i see your point, Astuur).

+1 for keeping the current state.

**Comment by bunnybot** [comment]
(by wolfsteinmetz)
I cannot really share your view here - both your views, so to speak :) - because I don't think
I am really giving up any controlling with this suggestion.
Obviously "letting the computer make decisons" is counterproductive and detrimental for the game.
But "waiting" or making this a "2-step process", is a mere technicality  (and a nuissance in my eyes).
All it does is keeping the player busy and distract from more rewarding controlling tasks.
I tend to think of myself as a ruler when playing WL.
What should be wrong with a ruler saying to some one of his staff: "Burn this down, and then build a Xy there."
In WL the machine is my staff - so that is how the idea came.
So your argumentation has not convinced me or change my mind in this case.
But you know my attitude - if you don't like it and feel it would be moving widelands into an undisirable direction,
I am not going to argue with the "makers".
Set it to "won't fix" or whatever.

**Comment by bunnybot** [comment]
(by victor-pelt)
just as a practical objection, what would it look like if you started building roads e.d. through a burning building.

**Comment by bunnybot** [comment]
(by hjd)
To me it has always made sense that I need to wait for the rest of the previous buildings to be completely burned up before I can build something new. Much the same way as I cannot build a new building where there is already a building or roads. (Clear the rubble to build anew). 

While I find waiting for the fires to die out, I see it as a part of the cycle of building/having/destroying a building. Building a building does not happen instantly so it makes sense that tearing it down doesn't either. This is also the most realistic option.

I sort of see why a planning feature would be useful to be able to plan the next building to put up. However, if introduced I think it would be strange to limit it to destroyed buildings. At that point, existing buildings (which should be replaced), or even planning which buildings to place in unconquered areas before expanding (which would then automatically be added).

To me, the current behaviour (while annoying from time to time) makes more sense. +1 to keeping it.

Victor: poor carriers! :(

**Comment by bunnybot** [comment]
(by wolfsteinmetz)
@ Victor:
I did not suggest to do anything  what you describe. I wanted to "place my orders" in a one time process, and then move on to something else. the actual rebuild of houses and roads should start after the completed burning (like it is now).

**Comment by bunnybot** [comment]
(by wolfsteinmetz)
@HJD
I can see what you have in mind,
but this one isn't meant to be the SirVer abandoned "land usage plans" sneaking in through the back door :)
http://wl.widelands.org/wiki/history/LandUsagePlans/changeset/1/

**Comment by bunnybot** [comment]
(by janitor)
[Expired for widelands because there has been no activity for 60 days.]

**Comment by bunnybot** [comment]
(by sirver)
I saw little support for this idea, I will therefore put it to rest. As always Astuur: if you feel you can rally more people to your cause reopen and relive this idea.
